### PR TITLE
fix(kafka): datahub-upgrade job

### PR DESCRIPTION
**Pull request by david-leifker**
fix(datahub-upgrade): add kafka health check bean
fix(kafka-setup): reduce memory usage

## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: datahub-upgrade/src/main/java/com/linkedin/datahub/upgrade/UpgradeCliApplication.java [hunk 1]
```diff
@@ -11,9 +11,13 @@
 
 @SuppressWarnings("checkstyle:HideUtilityClassConstructor")
 @SpringBootApplication(exclude = {ElasticsearchRestClientAutoConfiguration.class})
-@ComponentScan(basePackages = {"com.linkedin.gms.factory", "com.linkedin.datahub.upgrade.config"},
-    excludeFilters = {
-    @ComponentScan.Filter(type = FilterType.ASSIGNABLE_TYPE, classes = ScheduledAnalyticsFactory.class)})
+@ComponentScan(basePackages = {
+        "com.linkedin.gms.factory",
+        "com.linkedin.datahub.upgrade.config",
+        "com.linkedin.metadata.dao.producer"
```
Comment (david-leifker) [inline datahub-upgrade/src/main/java/com/linkedin/datahub/upgrade/UpgradeCliApplication.java:17]: This is the actual fix here for the kafka health check bean

**Review by RyanHolstien — APPROVED** [review]

File: docker/kafka-setup/kafka-setup.sh [hunk 2]
```diff
@@ -6,6 +6,8 @@
 
 : ${DATAHUB_ANALYTICS_ENABLED:=true}
 
+: ${KAFKA_HEAP_OPTS:=-Xmx64M}
```
Comment (pedro93) [inline docker/kafka-setup/kafka-setup.sh:9]: We shouldn’t hard code define the max heap size of the jvm. The docker image should instead use jre11 which by default consumes 75% of the available memory for the container.

Comment (david-leifker) [reply docker/kafka-setup/kafka-setup.sh:9]: @pedro93 - This is for running cli commands. It should not need 75% of the memory ever, settings fixed amount of memory per *cli command* is reasonable.

Comment (RyanHolstien) [reply docker/kafka-setup/kafka-setup.sh:9]: Default max is 25%, not 75%: https://docs.oracle.com/en/java/javase/11/gctuning/ergonomics.html#GUID-DA88B6A6-AF89-4423-95A6-BBCBD9FAE781

**Pull request merged**